### ISSUE TITLE
Customize the ObjectMapper used by DgsQueryExecutor in tests

### ISSUE THREAD

**Issue by srinivasankavitha** [Netflix/dgs-framework#1021]
The current DGSExecutor does not offer a way to customize the object mapper used for deserializing the responses, i.e. the various executeAndExtract (methods primarily used by tests).

**Comment by ClaudenirFreitas** [comment]
Well, you can check [Custom Object Mapper page](https://netflix.github.io/dgs/advanced/custom-object-mapper/).

**Comment by jakubjanecek** [comment]
I don't think this https://netflix.github.io/dgs/advanced/custom-object-mapper/ helps. I've tried it with no luck and the page says it's for serialization. It seems this global `ObjectMapper` https://github.com/Netflix/dgs-framework/blob/v4.10.2/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/BaseDgsQueryExecutor.kt#L48 is used for deserialization in tests and that can't be changed.

**Comment by srinivasankavitha** [comment]
That is correct, we need to implement support for this as mentioned in my original comment.

**Comment by heypko** [comment]
Did this issue ever get resolved?
